### PR TITLE
AB#12: Add new test cases of feature Search Functionality

### PR DESCRIPTION
**Pull request by LuanNguyenBao**
Test local result:
image

New changes:
1/ Add new clearByKeys() in BasePage.
2/ Add test cases of feature Search Functionality.

### CODE REVIEW

**Review by tuanna0308 — APPROVED** [review]
About the way to use the locator in the Test class, if you have time, you should move it to PageObject class, but if not, you can let it as now since the next sprint will test on another website.
I will merge this branch to the master branch, thanks for your hard work!

File: src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    @Test
+    @Description("Test description: Verify search with keyword 'Túi xách'")
+    @Story("Search Functionality")
+    public void verifySearchKeywordTuiXach() {
+        headerPage.sendKeyToElement(driver, By.cssSelector(SEARCH_TEXTBOX_CSS), "túi xách");
+        headerPage.clickToElement(driver, By.cssSelector(SEARCH_BUTTON_CSS));
+
+        String shopRelatedLabelExpected = "Shop liên quan đến \"túi xách\"";
```
Comment (khiemkao) [inline src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java:42]: should move lines 39 to 40 to PageObject by creating a new method "searchByKeyword(String keyword)"

File: src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        Assert.assertEquals(searchResultPage.getElementText(driver,
+                By.cssSelector(SHOP_RELATED_LABEL_CSS)), shopRelatedLabelExpected);
+
+        String searchResultLabelExpected = "Kết quả tìm kiếm cho từ khoá 'túi xách'";
+        Assert.assertEquals(searchResultPage.getElementText(driver,
+                By.cssSelector(SEARCH_RESULT_LABEL_CSS)), searchResultLabelExpected);
```
Comment (khiemkao) [inline src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java:48]: the PageUI should move to PageObject.
In test class, PageUI should not be present

File: src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    @Test
+    @Description("Test description: Verify search with keyword 'thời trang thu đông cho bé'")
+    @Story("Search Functionality")
+    public void verifySearchKeywordQuanAoThuDongChoBe() {
+        headerPage.sendKeyToElement(driver, By.cssSelector(SEARCH_TEXTBOX_CSS), "thời trang thu đông cho bé");
+        headerPage.clickToElement(driver, By.cssSelector(SEARCH_BUTTON_CSS));
```
Comment (khiemkao) [inline src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java:65]: we can reuse code from PageObject after creating method searchByKeyword

File: src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    @Test
+    @Description("Test description: Verify historical suggestion")
+    @Story("Search Functionality")
+    public void verifyHistoricalSuggestion() {
+        headerPage.clickToElement(driver, By.cssSelector(SEARCH_TEXTBOX_CSS));
+        Assert.assertTrue(searchResultPage.isElementDisplayed(driver, By.id(SEARCHBAR_LISTBOX_ID)));
+
+        headerPage.sendKeyToElement(driver, By.cssSelector(SEARCH_TEXTBOX_CSS), "túi xách");
+        headerPage.clickToElement(driver, By.cssSelector(SEARCH_BUTTON_CSS));
```
Comment (khiemkao) [inline src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java:93]: same as above

**Review by khiemkao — CHANGES_REQUESTED** [review]

**Review by tuanna0308 — APPROVED** [review]

**Review by khiemkao — APPROVED** [review]
looks good

**Pull request merged**